x-pack/plugins/alerts/server/invalidate_pending_api_keys/task.ts

```diff
@@ -0,0 +1,193 @@
+  config: Promise<AlertsConfig>,
+  securityPluginSetup?: SecurityPluginSetup
+) {
+  taskManager.registerTaskDefinitions({
+    [TASK_TYPE]: {
+      title: 'Invalidate alert API Keys',
+      createTaskRunner: taskRunner(logger, coreStartServices, config, securityPluginSetup),
+    },
+  });
+}
+
+function taskRunner(
+  logger: Logger,
+  coreStartServices: Promise<[CoreStart, AlertingPluginsStart, unknown]>,
+  config: Promise<AlertsConfig>,
+  securityPluginSetup?: SecurityPluginSetup
+) {
+  return ({ taskInstance }: RunContext) => {
+    const { state } = taskInstance;
+    return {
+      async run() {
+        let totalInvalidated = 0;
+        const configResult = await config;
+        try {
+          const [{ savedObjects }] = await coreStartServices;
+          const repository = savedObjects.createInternalRepository([
+            'api_key_pending_invalidation',
+          ]);
+          const configuredDelay = await configResult.invalidateApiKeysTask.removalDelay;
+          const delay = timePeriodBeforeDate(new Date(), configuredDelay).toISOString();
+
+          let hasApiKeysPendingInvalidation = true;
+          const PAGE_SIZE = 100;
+          do {
+            const apiKeysToInvalidate = await repository.find<InvalidatePendingApiKey>({
+              type: 'api_key_pending_invalidation',
+              filter: `api_key_pending_invalidation.attributes.createdAt <= "${delay}"`,
+              page: 1,
+              sortField: 'createdAt',
+              sortOrder: 'asc',
+              perPage: PAGE_SIZE,
+            });
+            totalInvalidated += await manageToInvalidateApiKeys(
+              logger,
+              repository,
+              apiKeysToInvalidate,
+              securityPluginSetup
+            );
+
+            hasApiKeysPendingInvalidation = apiKeysToInvalidate.total > 0;
+          } while (hasApiKeysPendingInvalidation);
+
+          return {
+            state: {
+              runs: (state.runs || 0) + 1,
+              total_invalidated: totalInvalidated,
+            },
+            schedule: {
+              interval: configResult.invalidateApiKeysTask.interval,
+            },
+          };
+        } catch (errMsg) {
+          logger.warn(`Error executing alerting apiKey invalidation task: ${errMsg}`);
+          return {
+            state: {
+              runs: (state.runs || 0) + 1,
+              total_invalidated: totalInvalidated,
+            },
+            schedule: {
+              interval: configResult.invalidateApiKeysTask.interval,
+            },
+          };
+        }
+      },
+    };
+  };
+}
+
+async function manageToInvalidateApiKeys(
+  logger: Logger,
+  repository: ISavedObjectsRepository,
+  apiKeysToInvalidate: SavedObjectsFindResponse<InvalidatePendingApiKey>,
+  securityPluginSetup?: SecurityPluginSetup
+) {
+  let totalInvalidated = 0;
+  for (const apiKeyObj of apiKeysToInvalidate.saved_objects) {
+    const response = await invalidateAPIKey(
```

Comment:
For a faster execution, you can do something like below to invalidate all the API keys at once instead of a for loop.

```
await Promise.all(apiKeysToInvalidate.saved_objects.map((apiKeyObj) => { ... }));
```